metricbeat/module/kafka/partition/partition.go

```diff
@@ -0,0 +1,97 @@
+package partition
+
+import (
+	"github.com/Shopify/sarama"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+// init registers the partition MetricSet with the central registry.
+func init() {
+	if err := mb.Registry.AddMetricSet("kafka", "partition", New); err != nil {
+		panic(err)
+	}
+}
+
+// MetricSet type defines all fields of the partition MetricSet
+type MetricSet struct {
+	mb.BaseMetricSet
+}
+
+// New create a new instance of the partition MetricSet
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+
+	config := struct{}{}
+
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	return &MetricSet{
+		BaseMetricSet: base,
+	}, nil
+}
+
+// Fetch partition stats list from kafka
+func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+
+	config := sarama.NewConfig()
+	config.Net.DialTimeout = m.Module().Config().Timeout
+	config.Net.ReadTimeout = m.Module().Config().Timeout
+	config.ClientID = "metricbeat"
+
+	client, err := sarama.NewClient([]string{m.Host()}, config)
```

Comment:
For the first version we should recommend to only use one Host as part of the docs but think long term on how we handle these kind of situations in metricbeat as this will not only apply to Kafka.